### PR TITLE
python3.pkgs.django-two-factor-auth: init at 1.15.1

### PR DESCRIPTION
**Pull request by DerDennisOP**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Mismatch between commit message and change.

Update of `google-cloud-asset` was done with #225609.

File: pkgs/development/python-modules/django-two-factor-auth/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    django-otp
+    django-phonenumber-field
+    qrcode
+  ];
+
+  # require internet connection
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Complete Two-Factor Authentication for Django.";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-two-factor-auth/default.nix:48]:
```suggestion
    description = "Complete Two-Factor Authentication for Django";
```

File: pkgs/development/python-modules/django-two-factor-auth/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+, django-otp
+, django-phonenumber-field
+, fetchFromGitHub
+, pythonOlder
+, qrcode
+, setuptools-scm
+}:
+
+buildPythonPackage rec {
+  pname = "django-payments";
```
Comment (natsukium) [inline pkgs/development/python-modules/django-two-factor-auth/default.nix:14]:
```suggestion
  pname = "django-two-factor-auth";
```

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by natsukium** [comment]
Would you add `pythonImportsCheck`? Otherwise, it seems fine to me.

File: pkgs/development/python-modules/django-two-factor-auth/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  ];
+
+  propagatedBuildInputs = [
+    django
+    django-formtools
+    django-otp
+    django-phonenumber-field
+    qrcode
+  ];
+
```
Comment (fabaff) [inline pkgs/development/python-modules/django-two-factor-auth/default.nix:44]: The [`extras_require`](https://github.com/jazzband/django-two-factor-auth/blob/master/setup.py#L21-L27) should be added by `passthru.optional-dependencies`.

File: pkgs/development/python-modules/django-two-factor-auth/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    owner = "jazzband";
+    repo = "django-two-factor-auth";
+    rev = "refs/tags/${version}";
+    hash = "sha256-+E6kSD00ChPiRLT2i43dNlVkbvuR1vKkbSZfD1Bf3qc=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "django-phonenumber-field>=1.1.0,<7" "django-phonenumber-field"
+  '';
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/django-two-factor-auth/default.nix]: thats likely not going to work with the setup.py used here

File: pkgs/development/python-modules/django-two-factor-auth/default.nix [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    ];
+    # yubikey = [
+    #   django-otp-yubikey
+    # ];
+    phonenumbers = [
+      phonenumbers
+    ];
+    # phonenumberslite = [
+    #   phonenumberslite
+    # ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/django-two-factor-auth/default.nix:68]:
```suggestion
    # ];
```

Comment (fabaff) [reply pkgs/development/python-modules/django-two-factor-auth/default.nix]: Thanks for the fix.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 225770` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.django-two-factor-auth</li>
    <li>python310Packages.django-two-factor-auth.dist</li>
    <li>python311Packages.django-two-factor-auth</li>
    <li>python311Packages.django-two-factor-auth.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/225770#issuecomment-1510083561)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-225770-2]$ cd results/python311Packages.django-two-factor-auth/lib/python3.11/site-packages/
 ...ew/pr-225770-2/results/python311Packages.django-two-factor-auth/lib/python3.11/site-packages]$ python3.11
Python 3.11.2 (main, Feb  7 2023, 13:52:42) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import two_factor
>>> two_factor.django.VERSION
(3, 2, 18, 'final', 0)
```

Looks good to me.

**Pull request merged**